Codigo/modNetwork.bas

```diff
@@ -252,45 +255,108 @@ End Sub
-        'Conexion activa? y es un usuario loggeado?
-102     If UserList(iUserIndex).ConnIDValida = 0 And UserList(iUserIndex).flags.UserLogged And currentTime - UserList(iUserIndex).Counters.TimeLastReset > DisconnectTimeout Then
-106         'mato los comercios seguros
-110         If UserList(iUserIndex).ComUsu.DestUsu.ArrayIndex > 0 Then
-112             If IsValidUserRef(UserList(iUserIndex).ComUsu.DestUsu) And UserList(UserList(iUserIndex).ComUsu.DestUsu.ArrayIndex).flags.UserLogged Then
-114                 If UserList(UserList(iUserIndex).ComUsu.DestUsu.ArrayIndex).ComUsu.DestUsu.ArrayIndex = iUserIndex Then
-116                     Call WriteConsoleMsg(UserList(iUserIndex).ComUsu.DestUsu.ArrayIndex, "Comercio cancelado por el otro usuario.", e_FontTypeNames.FONTTYPE_TALK)
-118                     Call FinComerciarUsu(UserList(iUserIndex).ComUsu.DestUsu.ArrayIndex)
+ForcedClose_Err:
+    Call TraceError(Err.Number, Err.Description, "modNetwork.KickConnection", Erl)
+End Sub
+
+Public Sub CheckDisconnectedUsers()
+    On Error GoTo CheckDisconnectedUsers_Err:
+100     If DisconnectTimeout <= 0 Then
+            Exit Sub
+        End If
+        Dim currentTime As Long
+        Dim iUserIndex As Integer
+102     currentTime = GetTickCount()
+104     For iUserIndex = 1 To MaxUsers
+106         With UserList(iUserIndex)
+                'Conexion activa? y es un usuario loggeado?
+108             If .ConnectionDetails.ConnIDValida = 0 And .flags.UserLogged Then
+110                 If .ConnectionDetails.ConnID > 0 Then
+112                     If currentTime - Mapping(.ConnectionDetails.ConnID).TimeLastReset > DisconnectTimeout Then
+                            'mato los comercios seguros
+114                         If .ComUsu.DestUsu.ArrayIndex > 0 Then
+116                             If IsValidUserRef(.ComUsu.DestUsu) And UserList(.ComUsu.DestUsu.ArrayIndex).flags.UserLogged Then
+118                                 If UserList(.ComUsu.DestUsu.ArrayIndex).ComUsu.DestUsu.ArrayIndex = iUserIndex Then
+120                                     Call WriteConsoleMsg(.ComUsu.DestUsu.ArrayIndex, "Comercio cancelado por el otro usuario.", e_FontTypeNames.FONTTYPE_TALK)
+122                                     Call FinComerciarUsu(.ComUsu.DestUsu.ArrayIndex)
+                                    End If
+                                End If
+124                             Call FinComerciarUsu(iUserIndex)
+                            End If
+126                         Call Cerrar_Usuario(iUserIndex, True)
+                        End If
                     End If
                 End If
-120             Call FinComerciarUsu(iUserIndex)
+           End With
+128     Next iUserIndex
+    
+        Exit Sub
+CheckDisconnectedUsers_Err:
+130     Call TraceError(Err.Number, Err.Description, "modNetwork.CheckDisconnectedUsers", Erl)
+End Sub
+
+Public Function MapConnectionToUser(ByVal ConnectionId As Long) As Integer
```

Comment:
Where is this called?

Reply:
this is called when loading/creating a character after pymmo tokens are validated

Reply:
But I mean there is no calls to this in this patch and you did not deleted the old function, how can this work? Where exactly in the code this is called since it's a new function cannot be called by existing code?

Reply:
So there are currently no code calling this actually?

Reply:
yes, there are calls in protocol at
1446 and 1486

Dim UserIndex As Integer
    UserIndex = MapConnectionToUser(ConnectionId)
    If UserIndex < 1 Then
        Call modSendData.SendToConnection(ConnectionId, PrepareShowMessageBox("No hay slot disponibles para el usuario."))
        Call KickConnection(ConnectionId)
        Exit Sub
    End If